File: backend/src/main/kotlin/com/example/backend/dtos/CommandeDto.kt
package com.example.backend.dtos

import com.fasterxml.jackson.annotation.JsonFormat
import org.springframework.data.domain.Page
import java.time.LocalDateTime

data class CommandeDTO(
  val id: Long?,
  val dateCreation: String?,
  val dateLivraison: String?,
  val fournisseurId: Long?,
  val produits: List<ProduitCommandeDTO>,
  val montantTotal: Double?,
  val etat: String?,
  val note: String?
)

data class CommandeNewDTO(
  val id: Int?,
  val nom: String?,
  val prix: Int?,
  val stock: Int?,
  val fournisseur: String?,
  val dateLivraison: String?,
  val datePeremption: String?,
  val quantiteStock: Int?,
  val prixAchat: Int?,
  val quantiteRestante: Int?,
)

data class ProduitCommandeDTO(
  val produitId: Long,
  val quantite: Int,
  val prixUnitaire: Double
)

data class CommandeRequest(
  val clientId: Long,
  val type: String,
  val employeId: Long,
  val fournisseurId: Long,
  val produits: List<ProduitCmdRequest>
)


data class ProduitCmdRequest(
  var codebarre: String?,
  var productId: Long?,
  var productCmdId: Long?,
  val id: Long?,
  val nom: String?,
  val quantite: Int? = 0,
  val quantiteRecu: Int? = 0,
  val uniteGratuite: Int? = 0,
//  @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
  val dateDePeremption: String?,
  val prixUnitaire: Double?,
  val prixVente: Double?,
  val prixAchat: Double?,
)

data class Produit(
  val id: Long?,
  val nom: String,
  val quantite: Int,
  val prixUnitaire: Double
)


data class CommandePageableCustomlDto(
  var content: Page<Map<String, Any?>?>?,
  var totalElements: Long?,
  var totalPages: Int?,
  var pageSize: Int?,
  var pageNumber: Int?,
  var totalAmountRecu: Double?,
  var totalAmountCommande: Double?,
  var totalQteRecu: Int?,
  var totalQteCommande: Int?,
)

File: backend_py/pharmaPython/app/schemas/commande_dto.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime


class ProduitCommandeDTO(BaseModel):
  produitId: int
  quantite: int
  prixUnitaire: float


class ProduitCmdRequest(BaseModel):
  codebarre: Optional[str] = None
  productId: Optional[int] = None
  productCmdId: Optional[int] = None
  id: Optional[int] = None
  nom: Optional[str] = None
  quantite: Optional[int] = 0
  qtiteCmd: Optional[int] = 0
  uniteGratuite: Optional[int] = 0
  datePeremption: Optional[str] = None  # ISO string ou datetime si tu préfères
  prixUnitaire: Optional[float] = None
  prixVente: Optional[float] = None
  prixAchat: Optional[float] = None


class ProduitCmdRuptureRequest(BaseModel):
  productId: Optional[int] = None
  producCmdtId: Optional[int] = None
  produitEnRayontId: Optional[int] = None
  reduction: Optional[int] = 0
  quantiteRestante: Optional[int] = 0
  datePeremption: Optional[str]  # ISO string ou datetime si tu préfères
  dateLivraison: Optional[str]  # ISO string ou datetime si tu préfères
  prix: Optional[float]
  prixAchat: Optional[float]


class Produit(BaseModel):
  id: Optional[int]
  nom: str
  quantite: int
  prixUnitaire: float


class CommandeDTO(BaseModel):
  id: Optional[int]
  dateCreation: Optional[str]  # ou datetime si tu veux parser
  dateLivraison: Optional[str]
  fournisseurId: Optional[int]
  produits: List[ProduitCommandeDTO]
  montantTotal: Optional[float]
  etat: Optional[str]
  note: Optional[str]


class CommandeNewDTO(BaseModel):
  id: Optional[int]
  nom: Optional[str]
  prix: Optional[int]
  stock: Optional[int]
  fournisseur: Optional[str]
  dateLivraison: Optional[str]
  datePeremption: Optional[str]
  quantiteStock: Optional[int]
  prixAchat: Optional[int]
  quantiteRestante: Optional[int]


class CommandeRequest(BaseModel):
  clientId: int
  type: str
  employeId: int
  fournisseurId: int
  produits: List[ProduitCmdRequest]


class CommandeRuptureRequest(BaseModel):
  type: str
  fournisseurId: int
  produits: List[ProduitCmdRuptureRequest]


class CommandePageableCustomlDto(BaseModel):
  content: Optional[List[Dict[str, Any]]]
  totalElements: Optional[int]
  totalPages: Optional[int]
  pageSize: Optional[int]
  pageNumber: Optional[int]
  totalAmountRecu: Optional[float]
  totalAmountCommande: Optional[float]
  totalQteRecu: Optional[int]
  totalQteCommande: Optional[int]
